File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] prefix = new int[nums.length];
        prefix[0] = 1;
        int subtotal = 1;
        for (int i = 0; i < nums.length - 1; i++) {
            subtotal *= nums[i];
            prefix[i + 1] = subtotal;
        }
        int[] suffix = new int[nums.length];
        suffix[nums.length - 1] = 1;
        subtotal = 1;
        for (int i = nums.length - 1; i > 0; i--) {
            subtotal *= nums[i];
            suffix[i - 1] = subtotal;
        }
        int[] ret = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            ret[i] = prefix[i] * suffix[i];
        }
        return ret;
    }
}  

File: Data Structures & Algorithms/find-pivot-index/submission-0.cs
public class Solution {
    public int PivotIndex(int[] nums) {
        List<int> prefix = new List<int>();
        int total = 0;

        for (int i = 0; i < nums.Length; i++) {
            prefix.Add(total);
            total += nums[i];
        }

        List<int> suffix = new List<int>();
        total = 0;

        for (int i = nums.Length - 1; i >= 0; i--) {
            suffix.Add(total);
            total += nums[i];
        }

        suffix.Reverse();

        for (int i = 0; i < nums.Length; i++) {
            if (prefix[i] == suffix[i]) {
                return i;
            }
        }

        return -1;
    }
}